src/Access/ContextAccess.cpp

```diff
@@ -204,6 +204,35 @@ namespace
             res.grant(AccessType::SELECT, DatabaseCatalog::INFORMATION_SCHEMA_UPPERCASE);
         }
 
+        /// There is overlap between AccessType sources and table engines, so the following code avoids user granting twice.
+        std::vector<std::tuple<AccessFlags, std::string>> source_and_table_engines = {
```

Comment:
```
static const std::vector<...> source_and_table_engines = ...
```
is faster because with `static const` it will need to initialize `source_and_table_engines` only once (no matter how many times `addImplicitAccessRights()` is called).